### ISSUE TITLE
Some error with bulk-add pronunciation

### ISSUE THREAD

**Issue by mjuzikfrik** [javdejong/nhk-pronunciation#22]
I get such an error with bulk-add pronunciation.

```
Error 
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed. 
If the issue only occurs when add-ons are enabled, please use the Tools>Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem. 
When you've discovered the add-on that is causing the problem, please report the issue on the add-ons section of our support site. 
Debug info:
Anki 2.1.14 (7b93e985) Python 3.6.7 Qt 5.12.1 PyQt 5.11.3
Platform: Windows 10
Flags: frz=True ao=True sv=1

Caught exception:
  File "C:\Users\em\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 475, in <lambda>
    a.triggered.connect(lambda: onRegenerate(browser))
  File "C:\Users\em\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 481, in onRegenerate
    regeneratePronunciations(browser.selectedNotes())
  File "C:\Users\em\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 583, in regeneratePronunciations
    note[dst] = getFormattedPronunciations(srcTxt)
  File "C:\Users\em\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 401, in getFormattedPronunciations
    prons = getPronunciations(expr, sanitize)
  File "C:\Users\em\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 387, in getPronunciations
    ret.update(getPronunciations(expr, sanitize))
  File "C:\Users\em\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 391, in getPronunciations
    for sub_expr in mecab_reader.reading(expr).split():
  File "C:\Users\em\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 212, in reading
    expr = self.mecab.stdout.readline().rstrip(b'\r\n').decode('euc-jp')
<class 'UnicodeDecodeError'>: 'euc_jp' codec can't decode byte 0xa4 in position 0: illegal multibyte sequence
```
Something to do with getting a pronunciation from phrases/sentences, I guess.

**Comment by SpongebobSquamirez** [comment]
In the near future this addon will likely be made obsolete by a different addon.
But in any case, "byte 0xa4 in position 0: illegal multibyte sequence" seems to be causing a problem. It's important to know what is trying to be read here. It's even possible that some very strange symbol found its way into your cards.

If you want a fix, I'd need you to give me a screenshot of the field, and preferably an actual copy of the text, causing the problem. You could send me your deck, but @javdejong is in charge of Anki 2.1 support, so I wouldn't be able to help you there.

If you're not sure how to locate the problematic card, you could proceed as follows:

1. Randomly select one half of your cards, and try to bulk add. 
2. Undo the bulk add if the error doesn't occur, select the other half, and redo bulk add.
3. Repeat the above process within the set of cards that gives the error, cutting the error-giving candidates in half each time until you've narrowed down the culprit to a single card (although it's certainly possible that multiple cards are problematic).

**Comment by mjuzikfrik** [comment]
The thing is there's no out-of-the-ordinary symbol there. A lot of them give me the error e.g. 二ラウンド, 三グラム, 五つほど.
Here's the deck.
[deck.zip](https://github.com/javdejong/nhk-pronunciation/files/3426685/deck.zip)

**Comment by SpongebobSquamirez** [comment]
I'll see if I can get the deck to work on Anki 2.0 (which is what I use). My first thought is that if it's not a bug with the program, the problem could be the spaces or the punctuation, which aren't Japanese. I'm wondering whether maybe they aren't compatible with the the codec that's currently used. Can you change the punctuation and spacing in one card and see if that fixes it?

There's also a possibility that there's some hidden, possibly malformed HTML in your field.

**Comment by mjuzikfrik** [comment]
What do you mean by punctuation? I have no spaces, commas or the like. There's a comma in one field but strangely enough it had no problems with that field. I've checked the problematic fields and there are no html tags.

**Comment by SpongebobSquamirez** [comment]
Wait, is 二ラウンド, 三グラム, 五つほど. 3 separate cards? That would explain a lot...

Then my only question is if your cards have any HTML you introduced yourself.

**Comment by mjuzikfrik** [comment]
No. I even exported the deck to plain text to see and there's no html.

**Comment by javdejong** [comment]
I am sorry to say that I cannot reproduce the error, with just about the same specs (Anki 2.1.14, and Windows 10). I imported the deck you provided, selected all cards in the browser, and bulk added pronunciation. I get no errors whatsoever... If you are still stuck with the error, I can try and add a try-except that will at least tell you for what expression things went wrong.

**Comment by honmashinsei** [comment]
I have a similar issue:

```
Error 
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed. 
If the issue only occurs when add-ons are enabled, please use the Tools>Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem. 
When you've discovered the add-on that is causing the problem, please report the issue on the add-ons section of our support site. 
Debug info:
Anki 2.1.13 (3ba55990) Python 3.6.7 Qt 5.12.1 PyQt 5.11.3
Platform: Windows 10
Flags: frz=True ao=True sv=1

Caught exception:
  File "aqt\reviewer.py", line 266, in <lambda>
  File "<decorator-gen-24>", line 2, in _answerCard
  File "anki\hooks.py", line 74, in decorator_wrapper
  File "anki\hooks.py", line 69, in repl
  File "aqt\reviewer.py", line 240, in _answerCard
  File "aqt\reviewer.py", line 96, in nextCard
  File "aqt\reviewer.py", line 168, in _showQuestion
  File "anki\cards.py", line 120, in q
  File "anki\cards.py", line 137, in _getQA
  File "anki\collection.py", line 589, in _renderQA
  File "anki\hooks.py", line 39, in runFilter
  File "C:\Users\honmashinsei\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 520, in add_pronunciation_once
    fields[dst] = getFormattedPronunciations(fields[src])
  File "C:\Users\honmashinsei\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 401, in getFormattedPronunciations
    prons = getPronunciations(expr, sanitize)
  File "C:\Users\honmashinsei\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 391, in getPronunciations
    for sub_expr in mecab_reader.reading(expr).split():
  File "C:\Users\honmashinsei\AppData\Roaming\Anki2\addons21\932119536\nhk_pronunciation.py", line 212, in reading
    expr = self.mecab.stdout.readline().rstrip(b'\r\n').decode('euc-jp')
<class 'UnicodeDecodeError'>: 'euc_jp' codec can't decode byte 0xa5 in position 2: illegal multibyte sequence
```

This started occurring after I updated the Japanese Support add-on today.

**Comment by javdejong** [comment]
Fixed with 09232150343f5bc912d3daa417de7843fa90276b (r30). Thanks to @TanukiDW in #23 for figuring out this was due to the Japanese Add-on being updated to UTF-8.
